dvc/cli.py

```diff
@@ -110,8 +111,26 @@ class DvcParser(argparse.ArgumentParser):
 
     def error(self, message, cmd_cls=None):  # pylint: disable=arguments-differ
         logger.error(message)
+        self._check_and_log_suggestions(message)
         _find_parser(self, cmd_cls)
 
+    def _check_and_log_suggestions(self, message):
+        cmd = regex_search(string=message, regex="[?=invalid choice: ]'(.*?)'")
```

Comment:
Is any method to get the false cmd directly instead of parsing it from the error message?

Reply:
In current state we are running the whole command before trying to get the error message and match failed command. We should do that before DVC tries to execute the command, mostly due to fact that this check will run upon every exception that DVC faces.

@naveen24691 in `cli.py` there is argument parsing part, I think that potential misuse detection should be happening somewhere there.

Reply:
@karajan1001  We can override `_check_value ` method. This method is called to check if command is supported or not. I can maintain an attribute to store the last checked value and use it here instead in case of error. But it will be difficult to figure it out for which reason this error method was called. We still need to check if "invalid choice" text is present in the string or not. We know something is wrong, only when error method is called with  a string message from within the `argparse.ArgumentParser `class.


@pared I think this `DvcParser.error` method is called only during the argument parser step. Command is executed post the successful parsing. Please correct me If I am wrong